unblob/processing.py

```diff
@@ -606,6 +606,9 @@ def _extract_chunk(self, file, chunk: ValidChunk):  # noqa: C901
         try:
             if result := chunk.extract(inpath, extract_dir):
                 extraction_reports.extend(result.reports)
+            elif chunk.is_whole_file and not self.config.keep_extracted_chunks:
+                logger.debug("Removing successfully processed file.", path=inpath)
+                inpath.unlink()
```

Comment:
It is pretty bad and misleading, that `chunk.extract` returns `ExtractResult | None`, it should always return a result, or raise an exception. I think `None` was kept for backward compatibility, when `ExtractResult` was introduced, but some/most `Handler`s are transformed to always return `ExtractResult`.

I think the above history led to this code, which translates to me "if there is a result, report problems, otherwise delete the file", so this code implies "result=something bad happened".

I think the intention here is if there is a report during extraction or an exception is raised, it is not a clean extraction, and should be kept, as it is potentially worth further looking into:

```suggestion
            result = chunk.extract(inpath, extract_dir)
            chunk_reports = result.reports if result else []
            extraction_reports.extend(chunk_reports)
            successfully_extracted = not chunk_reports

            if successfully_extracted and chunk.is_whole_file and not self.config.keep_extracted_chunks:
                logger.debug("Removing successfully processed file.", path=inpath)
                inpath.unlink()
```